### ISSUE TITLE
Autoconfiguration for OTel Instrumentation components

### ISSUE THREAD

**Issue by ThomasVitale** [spring-projects-experimental/spring-cloud-sleuth-otel#66]
**Is your feature request related to a problem? Please describe.**
Not a problem. I really like Spring Cloud Sleuth and the new OTel support is great. The OpenTelemetry Java Instrumentation project provides instrumentation for many common libraries (see [Standalone library instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/standalone-library-instrumentation.md)), such as Lettuce (Redis), OkHttp3, and HttpUrlConnection. I was wondering how to enhance Spring Cloud Sleuth OTel to support tracing through those instrumentation libraries.

**Describe the solution you'd like**
In Spring Cloud Sleuth, the Brave module provides additional features to trace extra libraries like gRPC and Lettuce. Could it be an option to do the same for OpenTelemetry and support more instrumented libraries? Or is there a simpler option to integrate an instrumentation library like `io.opentelemetry.instrumentation:opentelemetry-lettuce-5.1:1.3.0-alpha` with Spring Cloud Sleuth OTel?

**Describe alternatives you've considered**
At the moment, the only alternative to achieve full auto-instrumentation would be using the OpenTelemetry Java agent to trace all the libraries in my project.

**Comment by jonatan-ivanov** [comment]
There are two ways of instrumenting things with OTel:
1. using the agent
2. manual instrumentation (wrapping the client, registering a hook/interceptor, etc.)

With Sleuth-OTel, you need to follow the OTel docs and do the second option if there is no Sleuth instrumentation for the component you want to instrument or you want to explicitly use the OTel instrumentation. If you add this wrappers/interceptors/hooks, instrumentation should work out of the box with Sleuth-OTel, it's just not Sleuth will create the span but OTel. 

If you are referring to auto-configuring the OTel instrumentation components in Sleuth-OTel, so that you don't need to do this on your own, I think it is a valid request though at the moment we can't really do it, OTel tracing is rewriting the instrumentation API (see: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2713) so if we add auto-configuration, that code might be broken pretty fast (OTel tracing hasn't reached GA so it can still break the API).

**Comment by ThomasVitale** [comment]
@jonatan-ivanov thanks a lot for the detailed answer, now it's clearer to me. I was thinking about auto-configuration for the OTel instrumentation components in Sleuth OTel (similar to what it's been done with Brave), but of course it makes sense to wait for a more stable release.

When it reaches GA and if there's interest for the feature, I'll be available to help adding auto-configuration for extra OTel-instrumented libraries.

**Comment by jonatan-ivanov** [comment]
Great to hear, let me rename this issue so that it explicitly calls about that this is about auto-config.
I think we can create issues for the OTel components we write auto-config for one-by-one, if you create one, I can assign it to you and link this issue in.

I'm also thinking since OTel has already migrated 17 out of 98 components, it might be relatively safe to add autoconfig for those that are already done (last time I remember it was only 4).

@marcingrzejszczak WDYT?

**Comment by ThomasVitale** [comment]
Great, thank you! For starters, I have created an issue for Lettuce: https://github.com/spring-cloud-incubator/spring-cloud-sleuth-otel/issues/67 I'm available to work on that whenever you think it's a good time in relation to the progress in the OTel project.

**Comment by jonatan-ivanov** [comment]
@ThomasVitale I think based on @marcingrzejszczak's comment: https://github.com/spring-cloud-incubator/spring-cloud-sleuth-otel/issues/67#issuecomment-878146710, if you move that instrumentation from using Brave to use Sleuth API, you solved the issue in a way that is not blocked by the OTel instrumentation changes. Please feel free to submit a PR into Sleuth's  repo (also, please let us know if you have any questions, we are here to help).

**Comment by ThomasVitale** [comment]
@jonatan-ivanov I'll do that, thanks a lot!

**Comment by marcingrzejszczak** [comment]
Sleuth OTel is feature complete. We suggest migration to Micrometer Tracing - https://github.com/micrometer-metrics/tracing/wiki/Spring-Cloud-Sleuth-3.1-Migration-Guide
